classes/DirtEvent.sc

```diff
@@ -102,7 +102,7 @@ DirtEvent {
 			if(~legato.notNil) {
 				~delta * ~legato.value
 			} {
-				unitDuration = unitDuration ? ~delta;
+				unitDuration = (unitDuration ? ~delta) * (~timescale ? 1);
```

Comment:
it would be clearer and more efficient to write:
```
unitDuration = (unitDuration ? ~delta);
if(~timescale.notNil) { unitDuration = unitDuration * ~timescale };
```